pinot-common/src/main/java/org/apache/pinot/common/config/RoutingConfig.java

```diff
@@ -25,25 +25,48 @@
 
 
 public class RoutingConfig extends BaseJsonConfig {
+  public static final String PARTITION_AWARE_SEGMENT_PRUNER_TYPE = "partitionAware";
+  public static final String REPLICA_GROUP_INSTANCE_SELECTOR_TYPE = "replicaGroup";
+
+  @Deprecated
   public static final String ENABLE_DYNAMIC_COMPUTING_KEY = "enableDynamicComputing";
 
   private final String _routingTableBuilderName;
   private final Map<String, String> _routingTableBuilderOptions;
 
+  private final String _segmentPrunerType;
+  private final String _instanceSelectorType;
+
   @JsonCreator
   public RoutingConfig(@JsonProperty("routingTableBuilderName") @Nullable String routingTableBuilderName,
-      @JsonProperty("routingTableBuilderOptions") @Nullable Map<String, String> routingTableBuilderOptions) {
+      @JsonProperty("routingTableBuilderOptions") @Nullable Map<String, String> routingTableBuilderOptions,
+      @JsonProperty("segmentPrunerType") @Nullable String segmentPrunerType,
+      @JsonProperty("instanceSelectorType") @Nullable String instanceSelectorType) {
     _routingTableBuilderName = routingTableBuilderName;
     _routingTableBuilderOptions = routingTableBuilderOptions;
+    _segmentPrunerType = segmentPrunerType;
+    _instanceSelectorType = instanceSelectorType;
   }
 
+  @Deprecated
```

Comment:
Let's add `@Deprecated` with the PR that will hook this up with the real routing. Technically, we still depend on `routingTableBuilderName`.

Reply:
Good point